azurerm/resource_arm_application_gateway.go

```diff
@@ -1175,6 +1176,17 @@ func resourceArmApplicationGatewayCreateUpdate(d *schema.ResourceData, meta inte
 		gateway.ApplicationGatewayPropertiesFormat.WebApplicationFirewallConfiguration = expandApplicationGatewayWafConfig(d)
 	}
 
+	if stopApplicationGateway {
+		future, err := client.Stop(ctx, resGroup, name)
+		if err != nil {
```

Comment:
What will happen here if it's already stopped? 🤔

Reply:
Just checked by adding another stop section and it passed just fine. I think we're good!